pkg/framework/machinesets.go

```diff
@@ -285,6 +317,40 @@ func GetWorkerMachineSets(ctx context.Context, client runtimeclient.Client) ([]*
 	return result, nil
 }
 
+// getArchitectureFromMachineSetNodes returns the architecture of the nodes controlled by the given machineSet's machines.
+func getArchitectureFromMachineSetNodes(ctx context.Context, client runtimeclient.Client, machineSet *machinev1.MachineSet) (string, error) {
+	machines, err := GetMachines(ctx, client)
```

Comment:
i think this function could probably be simplified if we use the `GetNodesFromMachineSet` function, see https://github.com/openshift/cluster-api-actuator-pkg/blob/master/pkg/framework/nodes.go#L80